gh-136145: Define 'standard library' and 'stdlib' in the glossary (#136146)
python/cpython: Doc/glossary.rst

      and ending with double underscores.  Special methods are documented in
      :ref:`specialnames`.

   standard library

>       The collection of :term:`packages <package>`, :term:`modules <module>`
>       and :term:`extension modules <extension module>` distributed as a part
>       of the official Python interpreter package, or an individual member of
>       that collection.  The exact membership of the collection may vary based

Reviewer: As a noun, 'standard library' refers to the entire collecton.  As an adjective, it can be applied to any member.  Giving both definitions in one sentence is confusing.
```suggestion
      The collection of :term:`packages <package>`, :term:`modules <module>`
      and :term:`extension modules <extension module>` distributed as a part
      of the official Python interpreter package.  As an adjective, "a member of"
      that collection.  The exact membership of the collection may vary based
```

Author: I think I'd prefer to just leave out that sentence.

My intention was to cover the (admittedly rare) use of "standard library" to refer to an individual library, such as "... the `json` standard library."  But, I suppose that's still the adjective case, where "module" is implicit after "library" and really it's just superfluous.

>       on platform, available system packages, or other criteria.

Reviewer 2: I suggest adding:

```
See also :data:`sys.stdlib_module_names`.
```

      and ending with double underscores.  Special methods are documented in
      :ref:`specialnames`.

   standard library
      The collection of :term:`packages <package>`, :term:`modules <module>`
      and :term:`extension modules <extension module>` distributed as a part
      of the official Python interpreter package.  The exact membership of the

>       collection may vary based on platform, available system packages, or

Reviewer 3: ```suggestion
      collection may vary based on platform, available system libraries, or
```

(libraries can be found and loaded; packages are the distribution format that is installed)

      on platform, available system packages, or other criteria.

   stdlib

>       A synonym for :term:`standard library`.

Reviewer 4: Or? Synonym is also good. 
```suggestion
      An abbreviation of :term:`standard library`.
```

Author: I think I still lean towards 'synonym' for the simple fact that we have other "A synonym of :term:`...`." entries.  I'm fine with `An abbreviation of ...` if others prefer that distinction, though.

Reviewer 5: I think *abbreviation* is more correct here, since unlike the other three uses of *synonym*, this is not a proper word.

      issues such as incorrect types. See also :term:`type hints <type hint>`
      and the :mod:`typing` module.

   stdlib

>       An abbreviation of :term:`standard library`.

Reviewer 6: Should we hint that this isn't the appropriate term to use in CPython docs?
```suggestion
      An informal abbreviation of :term:`standard library`.
```

Reviewer 7: > Should we hint that this isn't the appropriate term to use in CPython docs?

Seems reasonable ... could use "unofficial" as an alternative to "informal".  (seems like this small change has gotten a little nitpicky, so I won't go on)

Reviewer 8: Both "informal" and "unofficial" works for me. But I agree that it's better with adjective.

Author: Given that we use the term in places like `sys.stdlib_module_names`, *is* it "informal" or "unofficial"?  There could be a case for the former, but I don't think we can say its "unofficial".
